core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -130,7 +131,22 @@ public void heartbeat() throws InterruptedException {
       return;
     }
 
-    TimeUnit.SECONDS.sleep(mQuietPeriodSeconds);
+    final Iterator<Long> jobIterator = mActiveJobToInodeID.keySet().iterator();
+
+    while (jobIterator.hasNext()) {
+      final Long jobId = jobIterator.next();
+      try {
+        final Status jobStatus = mReplicationHandler.getJobStatus(jobId);
+        if (jobStatus.isFinished()) {
```

Comment:
This includes failed jobs as well. so we will remove failed jobs and cause checker to rerun them. 
Or maybe this is the bug that you mentioned that you saw?

Reply:
It will absolutely rerun them. Is that bad?